boilerplate/openshift/custom-catalog-osd-operator/custom-catalog-build-push.sh

```diff
@@ -30,14 +30,48 @@ function build_catalog_image() {
   fi
 
   ${CONTAINER_ENGINE} pull ${bundle_image}
-  ${opm_local_executable} index add --bundles ${bundle_image} --tag ${image_tag} \
-    --container-tool ${CONTAINER_ENGINE_SHORT}
+  # some upstream bundles images are built specifying a "replaces" field, which means it builds on top of the 
+  # previous minor version. we need to check if the build fails for that reason and use another method
+  echo "building catalog image..."
 
+  # preserve logging output from command by duplicating the file descriptor
+  exec 5>&1
+  BUILD=$(${opm_local_executable} index add \
+    --bundles ${bundle_image} \
+    --tag ${image_tag} \
+    --container-tool ${CONTAINER_ENGINE_SHORT} 2>&1 | tee /dev/fd/5; exit ${PIPESTATUS[0]})
+  RC=$?
+  ERR_COUNT=$(echo $BUILD | grep -c 'non-existent replacement')
```

Comment:
Just a note here that `ERR_COUNT` will likely only ever be 0 or 1 unless you quote `"$BUILD"`. The way you're using it, that will still work, but jfyi.